### ISSUE TITLE
configOverrides is not overriding "processor" config key

### ISSUE THREAD

**Issue by kriansa** [stylelint/stylelint#1989]
## Issue description

I found a bug where documentation for `configOverrides` does not match actual implementation. On the docs, it says that one can override any configs coming either from a file or the `config` object. However, after several attempts to make the property `processors` work, I could see that only `rules` is being actually overwritten by `configOverrides`.

After digging into the code, I found [here](https://github.com/stylelint/stylelint/blob/master/src/augmentConfig.js#L75) that the config implementation does not allow many other settings to be actually overridden. You can see that in the current code, it only merges the object `configOverrides` after several operations in the config pipeline.

### How to reproduce

1. In a separate folder, run: **`npm install stylelint-config-primer stylelint-processor-arbitrary-tags`**

2. Create a file `index.js` with the following content, then run it with `node index.js`

```js
var stylelint = require('stylelint');
var formatter = require('stylelint/dist/formatters/stringFormatter').default;

var code = '<style>.badcss{ prop: val; }</style>';

stylelint.lint({
  code: code,

  config: {
    extends: 'stylelint-config-primer',
    processors: ['stylelint-processor-arbitrary-tags'], // Comment me!
  },

  configOverrides: {
    processors: ['stylelint-processor-arbitrary-tags'],
  }

}).then(function(resultObject) {
  console.log('SUCCESS', formatter(resultObject.results));
}).catch(function(e){
  console.error('ERROR', e);
});
```

### Expected behavior

Commenting line 11 (config.preprocessors) should make this script return the same output, because you have an override there.

### Current behavior

If you comment this line, the override is not being parsed and therefore the results will differ.

### Which version of stylelint are you using?

7.4.2

**Comment by davidtheclark** [comment]
@kriansa So that I understand this correctly — I think you're *not* saying that you're unable to override rules. In fact, you're only unable to override plugins and processors. Is that correct? If so, let's change the title of this issue.

**Comment by kriansa** [comment]
Hello @davidtheclark , I'm sorry if I wasn't very clear. It's actually the opposite. 

Currently, we are **not** able to override `plugins` or `processors`. The only config key I could successfully override is `rules`.

I've changed the title. Let me know if it's better.

Thanks!
